trio/_core/_multierror.py

```diff
@@ -429,7 +490,13 @@ def concat_tb(head, tail):
             )
         else:
 
-            def trio_show_traceback(self, etype, value, tb, tb_offset=None):
+            def trio_show_traceback(
+                self: Any,
+                etype: Any,
+                value: BaseException,
+                tb: TracebackType,
+                tb_offset: Any | None = None,
+            ) -> None:
                 # XX it would be better to integrate with IPython's fancy
```

Comment:
Examining the docs for [`set_custom_exc()`](https://ipython.readthedocs.io/en/stable/api/generated/IPython.core.interactiveshell.html#IPython.core.interactiveshell.InteractiveShell.set_custom_exc) and [`structured_traceback()`](https://ipython.readthedocs.io/en/stable/api/generated/IPython.core.ultratb.html#IPython.core.ultratb.SyntaxTB.structured_traceback), I think the types are:
```py
    self: IPython.InteractiveShell,
    etype: type[BaseException],
    tb_offset: int | None = None,
) -> str | None
```

Reply:
Changed in [e0f7a9c](https://github.com/python-trio/trio/pull/2742/commits/e0f7a9cac32647c303af28f101600ade0331a45e)